docs/source/testing/integration.rst

```diff
@@ -18,4 +18,164 @@
 Writing Integration Tests for NWChemEx
 ######################################
 
-TODO: Write me!!!!
+NWChemEx is a modular ecosystem designed with separation of concerns as a key
+design point. An example of this separation can be found with the 
+self-consistent field (SCF) code and the integrals code. These two components of
+NWX are linked by PluginPlay and SimDE and are intended to be used together, but
+neither is therefore explicitly required for the development of the other and 
+therefore reside in separate Plugins. The unit tests for these libraries are 
+intended to ensure basic functionality and correctness, which can usually be 
+accomplished with simple test data that allow the unit tests to run quickly. 
+With that said, the initial development and testing of the SCF becomes very 
+awkward when one is unable to easily acquire real integrals for real molecular 
+systems. Additionally, changes to the integrals code could have deleterious
+effects on the SCF code, which we would like to detect before merging. For these
+(and other reasons), it can be useful to implement integration tests to ensure
+the continued interoperability of the isolated components of the NWX stack.
+
+CMake for Integration Testing
+=============================
+
+The following code-block provides an example for how one can add the option for
+an integration test to a project that uses the NWX ecosystem.
+
+.. code-block:: CMake
+
+    # Include option
+    cmaize_option_list(
+        BUILD_INTEGRATION_TESTS ON "Build the integration tests?"
+    )
+
+    # How to build the integration tests
+    if("${BUILD_INTEGRATION_TESTS}")
+        include(nwx_pybind11)
+        # Set relevant test directories
+        set(CXX_INCLUDE_DIR /path/to/cxx/includes)
+        set(CXX_TEST_DIR /path/to/cxx/integration/tests)
+        set(PYTHON_TEST_DIR /path/to/python/integration/tests)
+
+        # Find or build needed dependencies
+        cmaize_find_or_build_dependency(
+            Catch2
+            URL github.com/catchorg/Catch2
+            BUILD_TARGET Catch2
+            FIND_TARGET Catch2::Catch2
+            VERSION ${NWX_CATCH2_VERSION}
+        )
+
+        cmaize_find_or_build_dependency(
+            chemcache
```

Comment:
As suggested by my previous comment, for integration testing I think we want to be pulling in NWChemEx. The problem with pulling in say Integrals to say test SCF is you'll have to duplicate all the module linking logic that happens in the NWChemEx (or at least will happen when experimental features are reintegrated).